UDV-Core/src/Utils/3DTiles/3DTilesBuildingUtils.js

```diff
@@ -0,0 +1,134 @@
+import { setTileVerticesColor, getBatchIdFromIntersection,
+  getBatchTableFromTile, 
+  getTileInLayer} from "./3DTilesUtils";
+
+/**
+ * Gets a building ID from an intersection. The intersecting object must
+ * be a "Mesh" object with a batch id.
+ * 
+ * @param {*} inter An intersection
+ */
+export function getBuildingIdFromIntersection(inter) {
+  let table = getBatchTableFromTile(inter.object);
+  let bid = getBatchIdFromIntersection(inter);
+  return table.content['cityobject.database_id'][bid];
+}
+
+/**
+ * Creates a Tile Building Info (TBI) dictionnary from a 3DTiles Layer.
+ * The TBI is an object containing associations between Building Ids and
+ * building-specific elements (mainly, the associated tile and the set of
+ * batch array indexes).
+ * 
+ * @param {*} layer The 3DTiles layer.
+ * @param {*} tbi An existing TBI for this layer. Tiles that are currently
+ * loaded in the layer will be added to the TBI if they're not already present.
+ * If no TBI is provided, a brand new one will be instantiated with currently
+ * loaded tiles.
+ * 
+ * @example
+ * let layer = view.getLayerById('3d-tiles-layer');
+ * //Fetch the TBI
+ * let tbi = getTilesBuildingInfo(layer);
+ * //Get a building ID from the mouse position
+ * let intersections = view.pickObjectsAt(mouseEvent, 5);
+ * let buildingId = getBuildingIdFromIntersection(
+ *                   getFirst3dObjectIntersection(intersections));
+ * //Display the building's infos
+ * console.log(tbi.buildings[buildingId]);
+ * 
+ * @example
+ * let layer = view.getLayerById('3d-tiles-layer');
+ * //Initialize the TBI
+ * let tbi = getTilesBuildingInfo(layer);
+ * //When the visible tiles change, update the TBI
+ * tbi = getTilesBuildingInfo(layer, tbi);
+ */
+export function getTilesBuildingInfo(layer, tbi = null) {
+  // Instantiate the TBI if it does not exist
+  if (!tbi) {
+    tbi = {};
+    tbi.totalTileCount = 0;
+    tbi.loadedTileCount = 0;
+    tbi.loadedTiles = {};
+    tbi.buildings = {};
+    tbi.tileset;
+  }
+  let tileIndex = layer.tileIndex;
+  let tileCount = tileIndex.index['1'].children.length;
```

Comment:
Changed to the following :

```js
let tileCount = Object.keys(tileIndex.index).length - 1;
```

tileIndex.index is not an array but a dictionnary so we have to use Object.keys to find its length. The minus 1 is because the root tile is listed in the tile index but has no geometrical information (hence no building), so I remove it because it will never be loaded in the TBI.